src/commands/mod.rs

```diff
@@ -2044,6 +2104,53 @@ fn combine_messages(
     Ok(description)
 }
 
+fn cmd_next(ui: &mut Ui, command: &CommandHelper, args: &NextArgs) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let child = workspace_command.resolve_single_rev("@+")?;
+    assert!(args.amount == 1 || args.amount > 1);
+    if args.edit {
+
+    }
+    // TODO(#NNN) We currently cannot deal with multiple children, which resulted from branches.    
+    todo!("Implement")
+    Ok(())
+}
+
+fn cmd_prev(ui: &mut Ui, command: &CommandHelper: args: &NextArgs) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let parent = workspace_command.resolve_single_rev("@-")?;
+    let edit = args.edit?;
+    assert!(args.amount == 1 || args.amount > 1);
+    // Handle the simple case of a basic `prev` call.
+    if args.amount == 1 {
+        // TODO(#NNN): Handle multiple parents.
+        if parent.parents().len() > 1 {
+            return Err(user_error("Revision has multiple parents,see issue #NNN for more info"));
+        }
+        workspace_command.check_rewritable(&parent)?;
+        let mut tx = workspace_command.start_transaction(format!("prev: {current} -> {parent}"))?;
+        if edit {
+            tx.edit(&parent).unwrap();
+            tx.finish(ui)?;
+            Ok(())
+        }
+        // Move the empty commit after the parent.
+        
+
+
+    }
+    assert!(args.amount > 1, "Expected more parents to traverse");
+
+    // TODO(#NNN): We currently cannot deal with multiple parents.
+    if parent.parents().len() > 1 {
+        return Err(user_error("Revision has multiple parents,see issue #NNN for more info"));
+    }
+    if edit {
+    }
+    todo!("implement")
+    Ok(())
```

Comment:
This is pretty unfinished now, but should also deal with the root commit so erroring out.